example/glue/CompactDelegate.hpp

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+ *
+ * (c) Copyright IBM Corp. 2017
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ *******************************************************************************/
+
+#ifndef COMPACTDELEGATE_HPP_
+#define COMPACTDELEGATE_HPP_
+
+#include "omrcfg.h" 
+#include "omrgcconsts.h"
+
+class MM_CompactScheme;
+class MM_EnvironmentBase; 
+class MM_MarkMap;
+
+#if defined(OMR_GC_MODRON_COMPACTION)
+
+/**
+ * Delegate class provides implementations for methods required for Collector Language Interface
+ */
+class MM_CompactDelegate
+{
+	/*
+	 * Data members
+	 */
+private:
+	OMR_VM *_omrVM;
+	MM_CompactScheme *_compactScheme;
+	MM_MarkMap *_markMap;
+
+protected:
+
+public:
+
+	/*
+	 * Function members
+	 */
+private:
+
+protected:
+
+public:
+	/**
+	 * Initialize the delegate.
+	 *
+	 * @param omrVM
+	 * @return true if delegate initialized successfully
+	 */
+	void
+	tearDown(MM_EnvironmentBase *env) { }
+
+	bool
+	initialize(MM_EnvironmentBase *env, OMR_VM *omrVM, MM_MarkMap *markMap, MM_CompactScheme *compactScheme)
+	{
```

Comment:
Can these not be parameters to the constructor instead?

Reply:
This was done to help keep the code to remain consistent with the other delegate classes.

Reply:
Delegate constructors have simple or nil parameter lists because they are usually instantiated inline with the owing class, which generally must be instantiated before the delegate can be initialized.